pkg/stringid/stringid.go

```diff
@@ -19,15 +19,25 @@ import (
 const shortLen = 12
 
 var (
-	validShortID = regexp.MustCompile("^[a-f0-9]{12}$")
-	validHex     = regexp.MustCompile(`^[a-f0-9]{64}$`)
+	validShortID *regexp.Regexp
+	validHex     *regexp.Regexp
 
 	rngLock sync.Mutex
 	rng     *rand.Rand // A RNG with seeding properties we control. It can only be accessed with randLock held.
+	once    sync.Once
```

Comment:
```suggestion
	onceRegex    sync.Once
```

Just to add some context to the variable.